src/backend/utils/mmgr/mcxt.c

```diff
@@ -705,6 +705,53 @@ MemoryContext_LogContextStats(uint64 siblingCount, uint64 allAllocated,
 	allAllocated, allFreed, contextName);
 }
 
+#ifdef EXTRA_DYNAMIC_MEMORY_DEBUG
+static int
+MemoryContextChunkStats_comparator(const void *l, const void *r)
+{
+	const MemoryContextChunkStats *l_chunk = *(MemoryContextChunkStats **)l;
+	const MemoryContextChunkStats *r_chunk = *(MemoryContextChunkStats **)r;
+
+	return r_chunk->bytes - l_chunk->bytes;
+}
+#endif
+
+static void
+MemoryContext_printTopListOfChunks(MemoryContext Context)
+{
+#ifdef EXTRA_DYNAMIC_MEMORY_DEBUG
+	Size full_table_size = CHUNKS_TABLE_SIZE;
+	MemoryContextChunkStats **chunks = malloc(full_table_size * sizeof(MemoryContextChunkStats *));
+	if (!chunks)
+		return;
+
+	Size ChunksCount = (*Context->methods.get_chunk_stats)(Context, &chunks, &full_table_size);
+
+	if (ChunksCount)
+	{
+		int show_count = ChunksCount < MAX_TOP_ALLOC_CHUNK_STATS ? ChunksCount : MAX_TOP_ALLOC_CHUNK_STATS;
+
+		qsort(chunks, ChunksCount, sizeof(MemoryContextChunkStats *), StandartChunkStats_comparator);
+		write_stderr("\tList of top %d (all %d) the biggest allocations\n", show_count, ChunksCount);
+		write_stderr("\tfunction, file:line, bytes, count, function_of_allocation\n");
+
+		for (int idx = 0; idx < show_count; idx++)
+		{
+			StandardChunkHeader *header = (StandardChunkHeader *) chunks[idx]->chunk;
+			write_stderr("\t%s, %s:%d, " UINT64_FORMAT " bytes, " UINT64_FORMAT ", %s\n",
+				header->parent_func, header->file, header->line,
+				chunks[idx]->bytes, chunks[idx]->count, header->exec_func);
+		}
+	}
+
+	for (int idx = 0; idx < ChunksCount; idx++)
+		free(chunks[idx]);
+	free(chunks);
+#else
+	return;
+#endif
+}
```

Comment:
1. Why did you open this function without `EXTRA_DYNAMIC_MEMORY_DEBUG` enabled?
2. Your last implementation to build top-N using insertion routine looked better by time complexity.